moma_mission/src/moma_mission/core/state_ros_control.py

```diff
@@ -14,12 +14,15 @@ def __init__(self, ns, outcomes=['Completed', 'Failure']):
         self.startlist = self.get_scoped_param("startlist", [])
         self.stoplist = self.get_scoped_param("stoplist", [])
         self.manager_namespace = self.get_scoped_param("manager_namespace", "")
-        self.ee_frame_param = self.get_scoped_param("ee_frame_param",
-                                                    "/{}/{}/tool_link".format(self.manager_namespace,
-                                                                              self.startlist[0])
-                                                    if len(self.startlist) > 0 else "")
-        self.ee_frame = rospy.get_param(self.ee_frame_param, None)
-        
+
+        # ???????
+        # self.ee_frame_param = self.get_scoped_param("ee_frame_param",
+        #                                             "/{}/{}/tool_link".format(self.manager_namespace,
+        #                                                                       self.startlist[0])
+        #                                             if len(self.startlist) > 0 else "")  -> does not work if the first controller in the list does not have this!
+        # self.ee_frame = rospy.get_param(self.ee_frame_param, None)
```

Comment:
I honestly dont like this solution. In theory we could have different controller which specify targets for a different frame (`panda_hand` vs `tool_frame` or even `panda_link<x>`). I know that the only use at the moment is `tool_frame` but this looks a bit like hard coding something and very difficult to inspect. 

Shall we instead have a global state machine namespace which contains common parameters? Right now we have a `scoped_param` but we can also have a `global_param` which instead would allow us to specify this shared parameters. At least they are more visible rather then hidden in the code. 

Also having the frame parsed from the controller would cause a dependency on the controller list. What if we load a controller only and this has no tool_frame param?

Reply:
> I honestly dont like this solution. In theory we could have different controller which specify targets for a different frame (panda_hand vs tool_frame or even panda_link<x>). I know that the only use at the moment is tool_frame but this looks a bit like hard coding something and very difficult to inspect.

Yes, but that's somehow intended at least in our solution. The MPC for example has its tool frame hard-coded and setting up a different one in the smach would cause unintended behavior (since it does not internally transform between these frames).

> Shall we instead have a global state machine namespace which contains common parameters? Right now we have a scoped_param but we can also have a global_param (still namespaced, e.g. /state_machine_name/param)which instead would allow us to specify this shared parameters. At least they are more visible rather then hidden in the code.

That would at least (to some degree) get rid of some of the hidden pitfalls. But it may be more useful to just keep `scoped_param` and make it fallback to the global param in case no local param for the current state is defined. So it would go like "lookup local"->"lookup global"->"fallback to default". And for the global params, it would probably make sense to have them defined in the same state-machine yaml.
Probably it makes sense to discuss this in person before implementation.

> Also having the frame parsed from the controller would cause a dependency on the controller list. What if we load a controller only and this has no tool_frame param?

In that case I would just set it to "None" or any other reasonable default (or fallback to the smach parameter in this case).

Reply:
Ok, right now this works. As you suggested, lets talk about this in person and have this merged now. Any solution would require non trivial changes that are better addressed in another PR.

Reply:
Ok.